docs/internals/optimizer.rst

```diff
@@ -385,6 +385,20 @@ The preprocessing components perform transformations to get the program
 into a certain normal form that is easier to work with. This normal
 form is kept during the rest of the optimization process.
 
+Some of the steps in this group should be considered prerequisites for all other components.
+Currently these are:
+
+- :ref:`disambiguator`
```

Comment:
above we write `Some steps depend on properties ensured by ...`, which does not list disambiguator but the rest of the four. For consistency, I'd also add it there (line 349, for some reason, gh does not want me to add a review comment there)